### ISSUE TITLE
OpenBar Version 22 crashes on Gnome 45.4: No signal connection 3 found

### ISSUE THREAD

**Issue by mkalinski** [neuromorph/openbar#18]
To reproduce:
1. Log in (OpenBar works normally).
2. Lock the screen.
3. Unlock the screen (OpenBar no longer works).

Logs:

```
gnome-shell[83581]: Extension openbar@neuromorph: Error: No signal connection 3 found

                                       Stack trace:
                                         _disconnect@resource:///org/gnome/gjs/modules/core/_signals.js:58:15
                                         disable@file:///home/mk/.local/share/gnome-shell/extensions/openbar@neuromorph/extension.js:901:77
                                         _callExtensionDisable@resource:///org/gnome/shell/ui/extensionSystem.js:202:32
                                         _onEnabledExtensionsChanged@resource:///org/gnome/shell/ui/extensionSystem.js:619:24
                                         async*_sessionUpdated@resource:///org/gnome/shell/ui/extensionSystem.js:799:20
                                         ExtensionManager/<@resource:///org/gnome/shell/ui/extensionSystem.js:44:18
                                         _callHandlers@resource:///org/gnome/gjs/modules/core/_signals.js:130:42
                                         _emit@resource:///org/gnome/gjs/modules/core/_signals.js:119:10
                                         _sync@resource:///org/gnome/shell/ui/sessionMode.js:212:14
                                         pushMode@resource:///org/gnome/shell/ui/sessionMode.js:170:14
                                         activate@resource:///org/gnome/shell/ui/screenShield.js:617:34
                                         lock@resource:///org/gnome/shell/ui/screenShield.js:684:14
                                         LockAsync@resource:///org/gnome/shell/ui/shellDBus.js:525:28
                                         _handleMethodCall@resource:///org/gnome/gjs/modules/core/overrides/Gio.js:373:35
                                         _wrapJSObject/<@resource:///org/gnome/gjs/modules/core/overrides/Gio.js:408:34
                                         @resource:///org/gnome/shell/ui/init.js:21:20
```

**Comment by neuromorph** [comment]
Thank you for reporting the issue!
I had to connect an external monitor and only then I was able to reproduce it. It is a bug introduced by the last update :(
I have fixed it here and that should work for you too. You may update now with following command.

`cd ~/.local/share/gnome-shell/extensions/openbar@neuromorph/; curl -LJO https://raw.githubusercontent.com/neuromorph/openbar/main/extension.js; cd`

Please let me know how it goes. I will then upload it as next update.
Thank you!

**Comment by mkalinski** [comment]
Funnily enough, I normally use an external monitor, but I observed this bug consistently on a laptop not connected to anything.

Looks like it's fixed for me on main, thanks. :+1:

**Comment by neuromorph** [comment]
Lol, from what I figured something was causing the primary monitor's index to change (can happen with multi-monitors) so I connect to one index while enabling the extension but try to disconnect from another during disable and it complaints about it. 
Anyway, I fixed the implementation to not run into that issue. Good to hear it worked!
Also uploaded it to extensions web; it's under review currently.
